pkg/hub/managedclusterset/controller.go

```diff
@@ -47,52 +49,34 @@ func NewManagedClusterSetController(
 	syncCtx := factory.NewSyncContext(controllerName, recorder)
 	clusterInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			if clusterSetName := getClusterSet(obj); len(clusterSetName) > 0 {
-				syncCtx.Queue().Add(clusterSetName)
-				return
-			}
+			syncCtx.Queue().Add(allClusterSetKey)
```

Comment:
I am thinking about building a index from cluster to []clusterset. So each time you know what is the existing clustersets for the cluster and what is the clustersets that are impacted.

Reply:
As we support labelSelector in current clusterset, the cluster's label change may impact all the clustersets. 
So we need to resync all clusterset when one cluster's label change.

Reply:
I do not it impact ALL clustersets, you still can figure out which clustersets is impacted, right?